rust/template/distributed_datalog/src/instantiate.rs

```diff
@@ -0,0 +1,250 @@
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::time::Duration;
+
+use differential_datalog::program::RelId;
+use differential_datalog::record::Record;
+use differential_datalog::DDlog;
+
+use crate::observe::Observable;
+use crate::schema::Config;
+use crate::schema::InOut;
+use crate::schema::Members;
+use crate::schema::Node;
+use crate::schema::RelInOut;
+use crate::tcp_channel::TcpSender;
+use crate::DDlogServer;
+
+type Outputs = BTreeMap<Node, HashSet<RelId>>;
+type Assignment = BTreeMap<Node, InOut>;
+
+/// Deduce the output streaming relations we require.
+///
+/// In a nutshell, this function deduces a mapping from all relations on
+/// a node to other nodes that have relations that have this relation as
+/// input. Unfortunately doing so is rather costly, as we ultimately
```

Comment:
Which is why we want to do this computation incrementally in the future.  If only there was a language that would let you do that ;)

Reply:
Haha :D One step at a time.